std/string.c3

```diff
@@ -0,0 +1,21 @@
+@require_once "assert.c3"
+@require_once "arithmetic.c3"
+
+typedef CString : { data : u8[&] }
+
+function length : (str: CString&) -> isize = {
+    let len : isize = 0;
+    const null : u8[1]& = "\0"; // TODO: add char
+
+    while str.data[len] != null[0] {
+        len = len + 1;
+    }
+
+    return len;
+}
+
+// TODO: this is kinda ugly, we need partial functions
```

Comment:
Hmm, took me a second to understand. This is a bad comment imo.

I think I wanted something like: `make<CString>()` rather than `make_cstring()`. But to do that we'd need to allow functions to have partial specializations (so the other generic arguments could be inferred).

Reply:
Ah, right. I remember you talking about that.